File: app/src/main/java/ru/e6atb/chat/CrashReportDispatcher.java
package ru.e6atb.chat;

import android.content.Context;

import org.json.JSONObject;

import java.util.List;

final class CrashReportDispatcher {
	private static final String LOG_BOT = "logbot";
	private static final int MAX_PER_PASS = 3;

	private CrashReportDispatcher() {
	}

	static synchronized int dispatch(Context context, MiniTaLib client) throws Exception {
		if (context == null || client == null || client.token().length() == 0) return 0;
		List<CrashReportStore.PendingReport> pending = CrashReportStore.pending(context);
		int sent = 0;
		for (CrashReportStore.PendingReport report : pending) {
			if (sent >= MAX_PER_PASS) break;
			JSONObject body = client.prepareMessage(
					LOG_BOT,
					report.text,
					"android-crash:" + report.id,
					true,
					0
			);
			client.sendPreparedMessage(body);
			CrashReportStore.remove(report);
			sent++;
		}
		return sent;
	}
}

File: app/src/main/kotlin/ru/e6atb/chat/CrashReportDispatcher.kt
package ru.e6atb.chat

import android.content.Context

internal object CrashReportDispatcher {
    private const val LOG_BOT = "logbot"
    private const val MAX_PER_PASS = 3

    @JvmStatic @Synchronized @Throws(Exception::class)
    fun dispatch(context: Context?, client: MST5?): Int {
        if (context == null || client == null || client.token().isEmpty()) return 0
        var sent = 0
        for (report in CrashReportStore.pending(context)) {
            if (sent >= MAX_PER_PASS) break
            val body = client.prepareMessage(LOG_BOT, report.text, "android-crash:${report.id}", true, 0)
            client.sendPreparedMessage(body)
            CrashReportStore.remove(report)
            sent++
        }
        return sent
    }
}
